tiledb/cf/netcdf_engine/_array_converters.py

```diff
@@ -39,10 +39,72 @@ class NetCDF4ArrayConverter(ArrayCreator):
              coordinate. Only allowed for sparse arrays.
     """
 
+    def _copy_to_array(
+        self,
+        netcdf_group: netCDF4.Group,
+        tiledb_array: tiledb.Array,
+        indexer: Tuple[slice, ...],
+        assigned_dim_values: Optional[Dict[str, Any]],
+        assigned_attr_values: Optional[Dict[str, np.ndarray]],
+    ):
+        """Copies data from a NetCDF group to a TileDB CF array.
+
+        Parameters:
+            netcdf_group: The NetCDF group to copy data from.
+            tiledb_uri: The TileDB array uri to copy data into.
+            tiledb_key: If not ``None``, the encryption key for the TileDB array.
+            tiledb_ctx: If not ``None``, the TileDB context wrapper for a TileDB
+                storage manager to use when opening the TileDB array.
+            assigned_dim_values: Mapping from dimension name to value for dimensions
+                that are not copied from the NetCDF group.
+            assigned_attr_values: Mapping from attribute name to numpy array of values
+                for attributes that are not copied from the NetCDF group.
+        """
+        netcdf_indexer = tuple(
+            index_slice
+            for index_slice, dim_creator in zip(indexer, self.domain_creator)
```

Comment:
Does this do the right thing if `indexer` doesn't have the same length as `self.domain_creator`?